junit-jupiter-engine/src/main/java/org/junit/jupiter/engine/descriptor/TestFactoryTestDescriptor.java

```diff
@@ -69,6 +70,18 @@ public boolean isLeaf() {
 	}
 
 	@Override
+	public JupiterEngineExecutionContext execute(JupiterEngineExecutionContext context,
```

Comment:
What do you think about invoking lifecycle callbacks around `@TestFactory` methods _and_ around each dynamic tests and making implementations aware of the type (container or test) they are invoked on ?
This could be done by adding an `isContainer:boolean` method to the `TestExtensionContext` like on the `TestDescriptor` interface.